### DISCUSSION TITLE
How to create a surface on a face?

### DISCUSSION THREAD

**Discussion by SyedZainAbbas** [CadQuery/cadquery#1238] [Q&A]
I am struggling to find an appropriate method to create a surface on a face. 

import cadquery as cq

box = (cq.Workplane().box(1,1,0.2).faces('>Z'))

I tried creating a sketch but only the sketch was visible, also tried interpPlate() but it did not give satisfactory results.
Please guide me on this issue. I will be thankful to you.

**Comment by adam-urbanczyk** [comment]
What do you want to model? What kind of surface are you trying to create?

**Reply by SyedZainAbbas** [reply]
I have created a frame for a semiconductor chip. Now, I want to give it some surface bodies (current in, current out) for the purpose of simulations.

This is the solid body I created.
![LF_1](https://user-images.githubusercontent.com/96447398/213913596-2fd8134d-a5c9-4922-a481-51b2c66d758d.PNG)

Now, I want to put some surface bodies in the blue areas however I am unable to find a suitable function for this purpose. I think 'interpPlate' method of Workplane class will work for this purpose but I could not wrap my head around it. Please explain this method or any other method that will do the job.
![LF_2](https://user-images.githubusercontent.com/96447398/213913521-878f1e46-932f-4fa5-b463-623c176f9474.PNG)

**Reply by adam-urbanczyk** [reply]
AFAICT you want to create 2D features. This use case is not really well supported, but you could do something like:

```python
bottom = cq.Workplane().rect(20,20).extrude(-1)
top = (
    bottom.faces('>Z')
    .pushPoints(((0,-5),(0,5)))
    .rect(15,3)
    .extrude(1, combine=False)
)

res = bottom.split(top)

show_object(res)
```
![afbeelding](https://user-images.githubusercontent.com/13981538/214367273-afc5401f-dbe7-4dfe-b4e8-eb65e78c522a.png)
